java/src/com/ibm/streamsx/topology/generator/spl/Preprocessor.java

```diff
@@ -68,4 +70,46 @@ public void compositeColocateIdUsage(List<JsonObject> composites) {
         for (JsonObject composite : composites)
             pePlacementPreprocess.compositeColocateIdUse(composite);
     }
+
+    @SuppressWarnings("serial")
+    private void relocateHashAdders(){
+        final Set<JsonObject> hashAdders = new HashSet<>();
+        // Firstly, find each hashAdder
+        GraphUtilities.visitOnce(GraphUtilities.findStarts(graph),
```

Comment:
Is there a reason to collect the list of hash adders first, rather than just relocate them as you discover them?

Reply:
probably is, due to to not being able to handle graph changes during the iteration.